public/jwpltx.js

```diff
@@ -98,27 +150,51 @@ window.jwpltx = window.jwpltx || {};
         } else {
           uri.q = 32;
         }
-        uri.ti = Math.round(uri.vd / uri.q);
+
+        uri.ti = 0;
         uri.pw = 0;
-        sendData('s');
 
+        // Initialize latest quantile to compare further quantiles with
+        lastQ = getLastVODQuantile(vp, vd, uri.q);
+        // Initial values to compare watched progress
+        lastVp = vp;
+
+        sendData('s');
         // monitor ticks for entering new quantile
       } else {
-        let pw = (Math.floor(vp / uri.ti) * 128) / uri.q;
-        if (pw != uri.pw) {
+        const pw = getProgressWatched(vp, vd);
+        const passedQ = getLastVODQuantile(vp, vd, uri.q);
+
+        // Total time watched since last t event.
+        timeWatched = timeWatched + (vp - lastVp);
+        lastVp = vp;
+
+        if (passedQ > lastQ) {
```

Comment:
It might be easier to follow if you keep this check in terms of quantiles (1-128) and check that pw (which is the current quantile) is greater than or equal to lastQ + 128 / uri.q

Reply:
These quantiles are ones we use to send `t` events, so for example if the video length is less than 300 but greater than 180, then we send 16 of them during the video, every videoLength / numberOfQuantiles seconds.

lastQ - is a last passed quantile. So for example if we are in the middle of the video, then this one if 8. When it changes to 9, then we send a new `t` event.

Reply:
So maybe we should keep `getLastVODQuantile ` name instead of `getLastChunkNumber` as it is more about quantiles and progress chunks.

Reply:
Using the 'chunk' terminology is just to differentiate quantiles (1-128) from what I think you called VODQuantiles (1-32.) I don't think that is a common term and it's a bit confusing to have 2 different types of quantiles. If you don't mind refactoring, I think just using quantiles (1-128) for everything will make it the most readable.

Reply:
I tried to do it this way:

```
// This way we can get quantile representation to compare with current progress
function getNextQuantile(progress, duration) {
    return Math.floor(getCurrentProgressQuantile(progress / duration) + MAX_DURATION_IN_QUANTILES / uri.q);
  }

// And then we can compare it with `pw`
if (pw >= nextQuantile) {
 ....
```

However we lose some accuracy and this case and player / ott flows start to behave a bit differently and some `ti` values differ as well.

Reply:
I have changed the names of functions and variables a bit to show where chunks and where quantiles are.

Reply:
There is certainly confusion around the quantile numbers, counts, values and how they are used interchangeably, that's why I want to be careful about how we describe it too.

I think your math in the first comment needs to factor q differently to make sure you're still looking at the right spacing for the quantiles and then it should work:

```
// This way we can get quantile representation to compare with current progress
function getNextTriggerQuantile(progress, duration) {
    return Math.ceil(progress/duration * uri.q) * MAX_DURATION_IN_QUANTILES / uri.q;
  }

// And then we can compare it with `pw`
if (pw >= nextQuantile) {
 ....

Reply:
Thank you! Seems to work properly now. I have changed both functions to use quantiles (0 - 128) now.

Reply:
Awesome! I will give it another look through!